fiftyone/core/collections.py

```diff
@@ -916,17 +916,28 @@ def _get_per_sample_frames_bytes(self):
         results = self._aggregate(pipeline=pipeline, attach_frames=True)
         return {str(r["_id"]): r["size_bytes"] for r in results}
 
+    def _get_first(self, limit=None, field=None, reverse=False):
+        pipeline = [{"$limit": limit if limit else 1}]
+        direction = -1 if reverse else 1
+        if field is None:
+            # Sort by _id if no field is specified to ensure first is deterministic
+            field = "_id"
+        pipeline.insert(0, {"$sort": {field: direction}})
+
+        cursor = self._aggregate(pipeline=pipeline)
+
+        if not cursor.alive:
+            raise ValueError("%s is empty" % self.__class__.__name__)
+        return cursor
+
     def first(self):
         """Returns the first sample in the collection.
 
         Returns:
             a :class:`fiftyone.core.sample.Sample` or
             :class:`fiftyone.core.sample.SampleView`
         """
-        try:
-            return next(iter(self))
-        except StopIteration:
-            raise ValueError("%s is empty" % self.__class__.__name__)
+        return next(self._get_first())
```

Comment:
This is returning dicts, not `Sample` or `SampleView` objects as promised. Which means that `DatasetView.first()`  is currently broken.

The existing pattern for situations like this where `Dataset.fcn` and `DatasetView.fcn` need different implementations is to raise `NotImplementedError` in `SampleCollection.fcn` and then implement the logic in the other two places.